### DISCUSSION TITLE
Axios navigation while doing request

### DISCUSSION THREAD

**Discussion by ithilbor** [axios/axios#5473] [Q&A]
Hi it's a few days that I'm into axios and I was wondering if after a user logs in, can I submit a request to the backend while the user can navigate the "Home page"? Can this be implemented with "Interceptors" or do I need to do it in another way?
Thanks in advance.

**Comment by jasonsaayman** [comment]
Yes, this is possible. Axios requests are asynchronous, so in a SPA you can start a request and navigate to the Home page without waiting for the request to finish.

Interceptors are usually not the right tool for this. They are better for cross-cutting behaviour like adding auth headers, refreshing tokens, logging, or handling global errors. The navigation/background-request flow should usually live in your login handler or application state logic.

Example:
```js
async function login(credentials) {
  const { data } = await axios.post('/login', credentials);
  // store token/user state
  localStorage.setItem('token', data.token);
  // start background request, but don't block navigation
  axios.post('/some-backend-task')
    .catch(error => {
      // handle/log failure somewhere appropriate
      console.error(error);
    });
  // navigate immediately
  navigate('/home');
}
```

One important detail: if your app cancels requests when a component unmounts, navigating away may abort the request. In that case, start the request from a place that survives route changes, such as a shared service/store, or avoid tying that request to the component lifecycle.

Also, if the user is doing a full page reload/navigation instead of SPA routing, the browser may cancel in-flight requests. For work that must be guaranteed, it is usually better to trigger it on the backend as part of the login flow or enqueue it server-side.
